src/ipred.rs

```diff
@@ -1,11 +1,50 @@
+use crate::include::common::attributes::ctz;
+use crate::include::common::bitdepth::AsPrimitive;
+use crate::include::common::bitdepth::BitDepth;
 use crate::include::common::bitdepth::DynPixel;
+use crate::include::common::bitdepth::BPC;
+use crate::include::common::intops::apply_sign;
+use crate::include::common::intops::iclip;
+use crate::include::dav1d::headers::Rav1dPixelLayout;
+use crate::src::levels::DC_128_PRED;
+use crate::src::levels::DC_PRED;
+use crate::src::levels::FILTER_PRED;
+use crate::src::levels::HOR_PRED;
+use crate::src::levels::LEFT_DC_PRED;
+use crate::src::levels::PAETH_PRED;
+use crate::src::levels::SMOOTH_H_PRED;
+use crate::src::levels::SMOOTH_PRED;
+use crate::src::levels::SMOOTH_V_PRED;
+use crate::src::levels::TOP_DC_PRED;
+use crate::src::levels::VERT_PRED;
+use crate::src::levels::Z1_PRED;
+use crate::src::levels::Z2_PRED;
+use crate::src::levels::Z3_PRED;
+use crate::src::tables::dav1d_dr_intra_derivative;
+use crate::src::tables::dav1d_filter_intra_taps;
+use crate::src::tables::dav1d_sm_weights;
+use cfg_if::cfg_if;
+use libc::memcpy;
 use libc::ptrdiff_t;
+use std::cmp;
 use std::ffi::c_int;
+use std::ffi::c_uint;
+use std::ffi::c_ulong;
+use std::ffi::c_ulonglong;
+use std::ffi::c_void;
+use std::slice;
 
-#[inline]
-pub unsafe fn get_upsample(wh: c_int, angle: c_int, is_sm: c_int) -> c_int {
-    return (angle < 40 && wh <= 16 >> is_sm) as c_int;
-}
+#[cfg(feature = "asm")]
+use crate::{include::common::bitdepth::bd_fn, src::cpu::rav1d_get_cpu_flags, src::cpu::CpuFlags};
+
+#[cfg(all(feature = "asm", target_arch = "aarch64"))]
+use ::to_method::To;
```

Comment:
Why does this path start with `::`? Could it just be `use to_method::To;`?

Reply:
Yeah, they're equivalent, but for `#[cfg]`ed imports, I usually group them into

```rust
#[cfg(something)]
use crate::{
   // all `crate::` imports
};

#[cfg(something)]
use ::{
   // all absolute imports
};
```

and then that reformats into this.  It's easier if we have to add more non-`crate::` imports to this later on.